src/Test/QuickCheck/Classes/Generic.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+#if MIN_VERSION_base(4,12,0)
+{-# LANGUAGE QuantifiedConstraints #-}
+#endif
+{-# OPTIONS_GHC -Wall #-}
+
+module Test.QuickCheck.Classes.Generic
+  (
+#if MIN_VERSION_base(4,6,0)
+    genericLaws
+  , genericLawsU
+#endif
+#if MIN_VERSION_base(4,12,0) 
+--  , generic1Laws
+#endif
+  ) where
+
+#if MIN_VERSION_base(4,6,0)
+import Control.Applicative
+import Data.Semigroup
+import Data.Monoid
+import GHC.Generics
+#if MIN_VERSION_base(4,9,0)
+import Data.Functor.Classes
+#endif
+import Data.Proxy (Proxy(Proxy))
+import Test.QuickCheck
+import Test.QuickCheck.Property (Property)
+
+import Test.QuickCheck.Classes.Common (Laws(..), Apply(..))
+
+-- | Tests the following properties:
+--
+-- [/Conjunction Idempotence/]
+--   @n .&. n ≡ n@
+-- [/Disjunction Idempotence/]
+--   @n .|. n ≡ n@
+-- [/Double Complement/]
+--   @complement (complement n) ≡ n@
+--
+-- /Note:/ This property test is only available when
+-- using @base-4.6@ or newer.
```

Comment:
Before `base-4.6` (GHC 7.6), `GHC.Generics` lived in `ghc-prim`, so you can even extend the support window further back if you depend on `ghc-prim`.

Reply:
Ok, will do.

Reply:
I can't seem to find on hackage a version of `ghc-prim` that exports the module `GHC.Generics`

Reply:
Alas, the GHC devs used to be pretty lax about uploading `ghc-prim` to Hackage, which means that the versions shipped with GHCs 7.2 through 7.4 (which expose `GHC.Generics`) aren't on Hackage.

Reply:
@andrewthad on his terminal showed me that it is ghc-0.2.0.0. I can just add this to the cabal file as a conditional dependency.

Reply:
resolved.